test/test_type_promotion.py

```diff
@@ -590,58 +590,61 @@ def test_transpose(self, device):
-            integral_quotient = torch.empty(5, device=device, dtype=dtype)
-            with self.assertRaises(RuntimeError):
-                torch.true_divide(dividend, divisor, out=integral_quotient)
-            with self.assertRaises(RuntimeError):
-                torch.true_divide(dividend, 2, out=integral_quotient)
-        else:
-            # Tests that requests for a floating quotient succeed
-            floating_quotient = torch.empty(5, device=device, dtype=dtype)
-            div_result = dividend / divisor
-            self.assertEqual(div_result,
-                             torch.true_divide(dividend, divisor, out=floating_quotient))
-            self.assertEqual(dividend / 2,
-                             torch.true_divide(dividend, 2, out=floating_quotient))
+    def test_div_promotion_out(self, device, dtype):
+        for op in (torch.div, torch.true_divide):
+            dividend = (torch.randn(5, device=device) * 100).to(dtype)
+            divisor = torch.arange(1, 6, device=device).to(dtype)
+
+            # Tests that requests for an integer quotient fail
+            if not dtype.is_floating_point:
+                integral_quotient = torch.empty(5, device=device, dtype=dtype)
+                with self.assertRaises(RuntimeError):
+                    op(dividend, divisor, out=integral_quotient)
+                with self.assertRaises(RuntimeError):
+                    op(dividend, 2, out=integral_quotient)
+            else:
+                # Tests that requests for a floating quotient succeed
+                floating_quotient = torch.empty(5, device=device, dtype=dtype)
+                div_result = dividend / divisor
+                self.assertEqual(div_result,
+                                 op(dividend, divisor, out=floating_quotient))
+                self.assertEqual(dividend / 2,
+                                 op(dividend, 2, out=floating_quotient))
 
     @dtypes(torch.float, torch.double,
             torch.bool, torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64)
-    def test_true_divide_inplace(self, device, dtype):
-        dividend = (torch.randn(5, device=device) * 100).to(dtype)
-        divisor = torch.arange(1, 6, device=device).to(dtype)
-
-        # Tests that requests for an integer quotient fail
-        if not dtype.is_floating_point:
-            with self.assertRaises(RuntimeError):
-                dividend.true_divide_(divisor)
-            with self.assertRaises(RuntimeError):
-                dividend.true_divide_(2)
-        else:
-            # Tests that requests for a floating quotient succeed
-            div_result = dividend.clone().div_(divisor)
-            self.assertEqual(div_result, dividend.clone().true_divide_(divisor))
-            self.assertEqual(dividend.clone().div_(2), dividend.clone().true_divide_(2))
+    def test_div_promotion_inplace(self, device, dtype):
+        for op in (torch.Tensor.div_, torch.Tensor.true_divide_):
```

Comment:
would be nicer if you could just assert div_ is true_divide_ eh?

Reply:
It would be. We could still set the functions equal in Python and C++.